x/wasm/keeper/keeper_test.go

```diff
@@ -2087,68 +2037,68 @@ func TestBuildContractAddress(t *testing.T) {
-			label:      "instance 1",
+			salt:       []byte("instance 1"),
 			creator:    "purple1nxvenxve42424242hwamhwamenxvenxvmhwamhwaamhwamhwlllsatsy6m",
 			expAddress: "purple1wde5zlcveuh79w37w5g244qu9xja3hgfulyfkjvkxvwvjzgd5l3qfraz3c",
 		},
-		"contract addr with different label": {
+		"contract addr with different salt": {
 			checksum:   fromHex("13A1FC994CC6D1C81B746EE0C0FF6F90043875E0BF1D9BE6B7D779FC978DC2A5"),
-			label:      "instance 2",
+			salt:       []byte("instance 2"),
 			creator:    "purple1nxvenxve42424242hwamhwamenxvenxvmhwamhwaamhwamhwlllsatsy6m",
 			expAddress: "purple1vae2kf0r3ehtq5q2jmfkg7wp4ckxwrw8dv4pvazz5nlzzu05lxzq878fa9",
 		},
 		"account addr with different checksum": {
 			checksum:   fromHex("1DA6C16DE2CBAF7AD8CBB66F0925BA33F5C278CB2491762D04658C1480EA229B"),
-			label:      "instance 1",
+			salt:       []byte("instance 1"),
 			creator:    "purple1nxvenxve42424242hwamhwamenxvenxvhxf2py",
 			expAddress: "purple1gmgvt9levtn52mpfal3gl5tv60f47zez3wgczrh5c9352sfm9crs47zt0k",
 		},
-		"account addr with different checksum and label": {
+		"account addr with different checksum and salt": {
 			checksum:   fromHex("1DA6C16DE2CBAF7AD8CBB66F0925BA33F5C278CB2491762D04658C1480EA229B"),
-			label:      "instance 2",
+			salt:       []byte("instance 2"),
 			creator:    "purple1nxvenxve42424242hwamhwamenxvenxvhxf2py",
 			expAddress: "purple13jrcqxknt05rhdxmegjzjel666yay6fj3xvfp6445k7a9q2km4wqa7ss34",
 		},
 		"contract addr with different checksum": {
 			checksum:   fromHex("1DA6C16DE2CBAF7AD8CBB66F0925BA33F5C278CB2491762D04658C1480EA229B"),
-			label:      "instance 1",
+			salt:       []byte("instance 1"),
 			creator:    "purple1nxvenxve42424242hwamhwamenxvenxvmhwamhwaamhwamhwlllsatsy6m",
 			expAddress: "purple1lu0lf6wmqeuwtrx93ptzvf4l0dyyz2vz6s8h5y9cj42fvhsmracq49pww9",
 		},
-		"contract addr with different checksum and label": {
+		"contract addr with different checksum and salt": {
 			checksum:   fromHex("1DA6C16DE2CBAF7AD8CBB66F0925BA33F5C278CB2491762D04658C1480EA229B"),
-			label:      "instance 2",
+			salt:       []byte("instance 2"),
 			creator:    "purple1nxvenxve42424242hwamhwamenxvenxvmhwamhwaamhwamhwlllsatsy6m",
 			expAddress: "purple1zmerc8a9ml2au29rq3knuu35fktef3akceurckr6pf370n0wku7sw3c9mj",
 		},
 		// regression tests
-		"min label size": {
+		"min salt size": {
 			checksum:   fromHex("13A1FC994CC6D1C81B746EE0C0FF6F90043875E0BF1D9BE6B7D779FC978DC2A5"),
-			label:      "x",
+			salt:       []byte("x"),
 			creator:    "purple1nxvenxve42424242hwamhwamenxvenxvhxf2py",
 			expAddress: "purple16pc8gt824lmp3dh2sxvttj0ykcs02n5p3ldswhv3j7y853gghlfq7mqeh9",
 		},
-		"max label size": {
+		"max salt size": {
 			checksum:   fromHex("13A1FC994CC6D1C81B746EE0C0FF6F90043875E0BF1D9BE6B7D779FC978DC2A5"),
-			label:      strings.Repeat("x", types.MaxLabelSize),
+			salt:       bytes.Repeat([]byte("x"), types.MaxLabelSize),
```

Comment:
If this tests passes then we should have a salt size check or assertion in BuildContractAddressPredictable because this is too long (due to bug here)

Reply:
There were no sanity checks in the address builder. The data can not be invalid when called by a message but I add some constraints when called as a library.